src/rust/catpowder/runtime/network.rs

```diff
@@ -61,16 +59,16 @@ impl NetworkRuntime for LinuxRuntime {
     }
 
     /// Receives a batch of [PacketBuf].
-    fn receive(&self) -> ArrayVec<Buffer, RECEIVE_BATCH_SIZE> {
+    fn receive(&self) -> ArrayVec<DemiBuffer, RECEIVE_BATCH_SIZE> {
         // 4096B buffer size chosen arbitrarily, seems fine for now.
         // This use-case is an example for MaybeUninit in the docs
         let mut out: [MaybeUninit<u8>; 4096] = [unsafe { MaybeUninit::uninit().assume_init() }; 4096];
         if let Ok((nbytes, _origin_addr)) = self.socket.borrow().recvfrom(&mut out[..]) {
-            let mut ret: ArrayVec<Buffer, RECEIVE_BATCH_SIZE> = ArrayVec::new();
+            let mut ret: ArrayVec<DemiBuffer, RECEIVE_BATCH_SIZE> = ArrayVec::new();
             unsafe {
                 let bytes: [u8; 4096] = mem::transmute::<[MaybeUninit<u8>; 4096], [u8; 4096]>(out);
-                let mut dbuf: Buffer = Buffer::Heap(DataBuffer::from_slice(&bytes));
-                dbuf.trim(4096 - nbytes);
+                let mut dbuf: DemiBuffer = DemiBuffer::from_slice(&bytes).expect("'bytes' should fit");
+                dbuf.trim(4096 - nbytes).expect("'bytes' <= 4096");
```

Comment:
This is out of the scope of this PR, but If possible, could you create a constant to hold this `4096` bytes?

Reply:
Yes, this is another thing that should be part of the coding standards (no unnamed inline constants that need to match another such value elsewhere).

Did you mean you'd like me to fix this in my PR, or file an Issue about it, or ???

Reply:
> Yes, this is another thing that should be part of the coding standards (no unnamed inline constants that need to match another such value elsewhere).

Agree. @anandbonde could you note this down for out next discussion?

> Did you mean you'd like me to fix this in my PR, or file an Issue about it, or ???

Yes, could you fix this in this PR if possible?

Reply:
This pre-existing problem has been fixed.

Reply:
Also, I added a description of this issue to the "Coding Standards" Channel in Teams and asked Anand to add it to the agenda for the next meeting.